### PR TITLE
Handling for Non Numeric Costs

### PR DESCRIPTION
**Pull request by bordoni**
_Ref:_ [C#39041](https://central.tri.be/issues/39041)
_Rel:_ moderntribe/events-filterbar#52

### CODE REVIEW

File: src/Tribe/Cost_Utils.php [hunk 1]
```diff
@@ -208,7 +208,7 @@ protected function get_cost_by_func( $costs = null, $function = 'max' ) {
 		}
 
 		// use a regular expression instead of is_numeric
-		if ( ! preg_match( $this->get_cost_regex(), $cost ) ) {
+		if ( empty( $cost ) ) {
```
Comment (barryhughes) [inline src/Tribe/Cost_Utils.php]: In changing this line we should also update the comment above it (or remove it).

File: src/Tribe/Cost_Utils.php [hunk 2]
```diff
@@ -310,8 +310,15 @@ public function parse_cost_range( $costs, $max_decimals = null ) {
...
-			// Creates a Well Balanced Index that will perform good on a Key Sorting method
-			$index = str_replace( array( '.', ',' ), '', number_format( str_replace( $this->get_separators(), '.', $cost ), $max ) );
+			$numeric_cost = str_replace( $this->get_separators(), '.', $cost );
+
+			if ( is_numeric( $numeric_cost ) ) {
+				// Creates a Well Balanced Index that will perform good on a Key Sorting method
+				$index = str_replace( array( '.', ',' ), '', number_format( $numeric_cost, $max ) );
+			} else {
+				//
+				$index = sanitize_title( $numeric_cost );
```
Comment (barryhughes) [inline src/Tribe/Cost_Utils.php]: * Does this have potential to mean some indices will be 'well balanced' and some not? Does that have implications for use of min() and max() further on in the request?
* Comment is empty: add something or remove it!

File: src/Tribe/Importer/Admin_Page.php [hunk 3]
```diff
@@ -159,19 +160,24 @@ public function render_csv_tab() {
 						$header[] = $letter++;
 					}
 				}
+
 				$import_type = get_option( 'tribe_events_import_type' );
 				$messages = $this->errors;
 				include Tribe__Events__Importer__Plugin::path( 'src/io/csv/admin-views/columns.php' );
+
```
Comment (barryhughes) [inline src/Tribe/Importer/Admin_Page.php]: These are just misc formatting tweaks unrelated to the main thrust of the PR right? How come the extra line before breaking here, but not in the case of the subsequent case statements?

File: vendor/tickets [hunk 4]
```diff
@@ -1 +1 @@
-Subproject commit 4c327a4537eb3f183e56b7c59455d531804ccc20
+Subproject commit fa22e4a2e823f7bf3af84dcb75c038cda8133785
```
Comment (barryhughes) [inline vendor/tickets]: This moves the submodule back several commits - accidental?

Comment (bordoni) [inline vendor/tickets]: Thanks! Thats acidental

File: src/Tribe/Importer/Admin_Page.php (hunk 3 above)
Comment (bordoni) [inline src/Tribe/Importer/Admin_Page.php]: I will just remove that, I was testing something there.

File: src/Tribe/Cost_Utils.php (hunk 2 above)
Comment (bordoni) [inline src/Tribe/Cost_Utils.php]: So on all my testing it seems to be working well, with the Min and Max conditionals.

Comment (barryhughes) [inline src/Tribe/Cost_Utils.php]: Totally trust your judgement, just jumped out at me so I wanted to ask :+1:

**Comment by barryhughes** [comment]
:finnadie:

**Pull request merged**